### PR TITLE
metrics: add valuation metrics

### LINKED ISSUE #258: Add cost metric
Add cost metric(s) to the `metrics` subpackage. Can focus first on the "simple" case where a user provides a cost per error value (e.g. `$/MW`), either a single value or a (potentiall) time-series corresponding to the forecasts/observations.

### PR DESCRIPTION
**Pull request by dplarson**
- [x] Closes #258 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

This PR is focused on adding some forecast valuation metrics for quantifying the monetary cost of forecast errors. Note: the name (`solarforecastarbiter.metrics.valuation`) was chosen to avoid namespace conflicts with the (more commonly used terms) `cost` and `value`

### CODE REVIEW

**Comment by dplarson** [comment]
@wholmgren This PR on adding valuation metrics is now ready for review. Right now, it just covers adding `fixed_error_cost(obs, fx, cost)` which implements computing the total cost of misforecasts based on a (scalar) fixed cost per error (e.g. USD per MW of error).

File: solarforecastarbiter/metrics/valuation.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    -------
+    total_cost : float
+        Total cost (e.g. USD).
+
+    Examples
+    --------
+
+    Forecast power [kW], cost: 10 USD per kW
+    >>> fx = np.array([1, 2, 3])
+    >>> obs = np.array([1, 3, 4])
```
Comment (wholmgren) [inline solarforecastarbiter/metrics/valuation.py]:
```suggestion
    >>> obs = np.array([1, 1, 4])
```

Comment (wholmgren) [reply solarforecastarbiter/metrics/valuation.py]: demonstrates `abs`

File: solarforecastarbiter/metrics/valuation.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    >>> cost = 10  # 10 USD per kW
+    >>> fixed_error_cost(fx, obs, cost)
+    20
+
+    """
+
+    error = np.abs(fx - obs)
+    total_cost = np.sum(error * cost)
```
Comment (wholmgren) [inline solarforecastarbiter/metrics/valuation.py]:
```suggestion
    total_cost = np.sum(error) * cost  # more efficient than sum(error * cost)
```

File: solarforecastarbiter/metrics/valuation.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+"""Metrics for quantifying the (monetary) value of forecasts."""
```
Comment (wholmgren) [inline solarforecastarbiter/metrics/valuation.py]:
```suggestion
"""Metrics for quantifying the monetary value of forecasts."""
```

File: solarforecastarbiter/metrics/valuation.py (hunk 2 above)
Comment (dplarson) [reply solarforecastarbiter/metrics/valuation.py]: Revised as suggested.

File: solarforecastarbiter/metrics/valuation.py (hunk 1 above)
Comment (dplarson) [reply solarforecastarbiter/metrics/valuation.py]: Good suggestion. Revised accordingly.

File: solarforecastarbiter/metrics/valuation.py (hunk 3 above)
Comment (dplarson) [reply solarforecastarbiter/metrics/valuation.py]: Revised as suggested.

**Comment by dplarson** [comment]
Do we want to add a function for comparing the cost of two forecasts? For example: a function that computes `value = cost_fx - cost_ref` (where `cost_fx` is the cost for a "new" forecast and `cost_ref` is the cost of a reference forecast)? Or is that too simple to warrant a separate function?

**Comment by dplarson** [comment]
In that case, this PR is ready to be merged.

**Pull request merged**